### PR TITLE
Autogenerate Display impls from doc comments

### PR DESCRIPTION
**Pull request by mystor**
This also includes Display implementations for `iface::Ide`, however they're quite verbose to match the [OSDev Wiki](https://wiki.osdev.org/Pci#Class_Codes).

I'm unfortunately not super satisfied with the display format. Only the first attribute on each variant is checked, and must be a doc comment. See tests for examples.

### CODE REVIEW

File: pci/src/class.rs [hunk 1]
```diff
@@ -292,4 +413,18 @@ mod test {
             Ok(Class::MassStorage(MassStorage::Sata(iface::Sata::Achi1))),
         );
     }
+
+    #[test]
+    fn test_display() {
+        assert_eq!(
+            Class::MassStorage(MassStorage::Sata(iface::Sata::Achi1)).to_string(),
+            "Mass Storage Controller : Serial ATA : AHCI 1.0"
```
Comment (hawkw) [inline pci/src/class.rs]: I'd prefer to not put the spaces on both sides of the colon (which feels a little weird IMO). What do you think about maybe something like
```rust
"Mass Storage Controller: Serial ATA (AHCI 1.0)"
```
we could also consider using the alt format specifier, or the precision specifier, to control how much information is formatted?

**Comment by hawkw** [comment]
@mystor shall i merge this now & we can tweak the formatting in a follow-up, or do you want to do it now?

**Comment by mystor** [comment]
I'm fine with tweaking formatting in a follow-up

**Comment by mystor** [comment]
@hawkw I added the changes you suggested, removing the space before the `:`, and supporting precision specifiers. Feel free to merge if you're satisfied.

**Pull request merged**